spec/services/processors/blog_metrics_updater_spec.rb

```diff
@@ -0,0 +1,183 @@
+require 'rails_helper'
+
+describe Processors::BlogMetricsUpdater do
+  describe '.call' do
+    let(:technology) { create(:technology) }
+    let!(:blog_post) { create(:blog_post, technology: technology) }
+    let(:blog_post_views_payload) { create(:blog_post_views_payload) }
+
+    before do
+      stub_blog_post_views_response(blog_post.blog_id, blog_post_views_payload)
+    end
+
+    it 'updates blog post visits metrics' do
+      described_class.call
+
+      metric = Metric.find_by!(
+        ownable: blog_post,
+        interval: Metric.intervals[:monthly],
+        name: Metric.names[:blog_visits]
+      )
+
+      expect(metric).to be_present
+    end
+
+    it 'updates technologies visits metrics' do
+      described_class.call
+
+      metric = Metric.find_by!(
+        ownable: technology,
+        interval: Metric.intervals[:monthly],
+        name: Metric.names[:blog_visits]
+      )
+
+      expect(metric).to be_present
+    end
+  end
+
+  describe '#update_blog_post_visits_metrics' do
```

Comment:
I think testing private methods is not a good practice. Perhaps we can add these tests under the describe for `.call` and group them with contexts.

Reply:
I see what you say. I wanted to tackle this on a more unit-testing approach, so that it is better isolated.
If I were to move them to the `'.call'` describe, I would mess up the setup I'm creating in `'#update_technologies_visits_metrics'`, because the other method I run in `call` (`update_blog_post_visits_metrics`) would interfere on it (it creates the blog post visits metrics I use there to create the technology ones).

I think the alternative could be to move both and to create a separate processor for updating the technologies visits metrics (which may be the right move anyway), and test it in isolation in its own test suite.
What do you think?

Reply:
I don't think I fully understood, why would it interfere? If your blog post visits metrics are already present before you run `.call`, then `update_blog_post_visits_metrics` shouldn't do anything right?

Reply:
Oh, you're right, it shouldn't! Thank you! I'll move them both, then

Reply:
Both setups and assertions did still interfere with each other, so it wasn't that straightforward. So I ended up creating a separate processor for technologies, as I think I would end up doing it in the future anyway.